src/Utilities/TaggedTuple.hpp

```diff
@@ -776,4 +776,44 @@ std::ostream& operator<<(std::ostream& os,
   return os << ")";
 }
 
+namespace TaggedTuple_detail {
+
+template <typename F, typename... Tags, typename... ApplyTags>
+constexpr decltype(auto) apply_impl(F&& f, const TaggedTuple<Tags...>& t,
+                                    tmpl::list<ApplyTags...> /* meta */) {
+  return std::forward<F>(f)(get<ApplyTags>(t)...);
+}
+
+}  // namespace TaggedTuple_detail
+
+// @{
+/*!
+ * \ingroup UtilitiesGroup
+ * \brief Invoke `f` with the `ApplyTags` taken from `t` expanded in a parameter
+ * pack
+ *
+ * `ApplyTags` defaults to the full list of tags in `t`.
+ *
+ * Here is an example how to use the function:
+ *
+ * \snippet Test_TaggedTuple.cpp expand_tuple_example
+ *
+ * This is the function being called in the above example:
+ *
+ * \snippet Test_TaggedTuple.cpp expand_tuple_example_function
+ *
+ * \see cpp17::apply
+ */
+template <typename ApplyTags, typename F, typename... Tags>
+constexpr decltype(auto) apply(F&& f, const TaggedTuple<Tags...>& t) {
+  return TaggedTuple_detail::apply_impl(std::forward<F>(f), t, ApplyTags{});
```

Comment:
Please drop this commit. I don't like the DataBox functionality and don't like this either.

Reply:
It's `cpp17::apply` for a TaggedTuple.. what's there to dislike :D https://en.cppreference.com/w/cpp/utility/apply

Reply:
It's for expanding items from a TaggedTuple into a parameter pack

Reply:
ah okay, I was worried it was the hwole `::arguments_tags` business again... Do we want/have the tuple version in `Utilities/Tuples.hpp`?

Reply:
yea I added that in #2083

Reply:
also please read the function first before requesting to drop a commit, and give a reason other than "I don't like it". And I will make a better effort to test my branches on CI before submitting the PR :)

Reply:
Oh, wait, no this doesn't do what you say it does... You specify `ApplyTags`, which is the part I don't like...  I'll need to think about it more, but it certainly doesn't do what `cpp17::apply` does, which passes _all_  arguments of the tuple forward. You have that one below. However, that's the only one that exists for `std::tuple`.

Reply:
yea I figured since the TaggedTuple has this way of identifying its element by tags, there can be an overload of `apply` that selects a subset and/or ordering of tags to expand into the function

Reply:
Having thought about this more, I'm fine with it. Marking comment as resolved :)